### DISCUSSION TITLE
how to disable shell integration on zsh over ssh ?

### DISCUSSION THREAD

**Discussion by ystyle** [wezterm/wezterm#4699] [Q&A]
### What Operating System(s) are you running on?

Linux Wayland

### Which Wayland compositor or X11 Window manager(s) are you using?

hyprland

### WezTerm version

none

### Ask your question!

how to disable shell integration on zsh over ssh ? (I use windterm or windows terminal on windows, to connect my laptop)
![image](https://github.com/wez/wezterm/assets/4478635/af1d4195-49d8-4a83-b92a-1fab70756540)

**Comment by nekowinston** [comment]
You can exit loading the shell extension early via `WEZTERM_SHELL_SKIP_ALL`
https://github.com/wez/wezterm/blob/e3cd2e93d0ee5f3af7f3fe0af86ffad0cf8c7ea8/assets/shell-integration/wezterm.sh#L6-L11

You could set it early in your host bashrc/zshrc/etc, when an SSH connection is active:
```bash
[ -n "$SSH_TTY" ] && export WEZTERM_SHELL_SKIP_ALL=1
```

**Comment by ystyle** [comment]
After trying it, you should modify the first line of the following file
```shell
# cat /etc/zsh/zprofile 
[ -n "$SSH_TTY" ] && export WEZTERM_SHELL_SKIP_ALL=1
emulate sh -c 'source /etc/profile'
````

**Comment by jmpolom** [comment]
Is this still a valid way to disable shell integration in wezterm? Where is `WEZTERM_SHELL_SKIP_ALL=1` documented?

**Reply by nekowinston** [reply]
Yes - if it's set to `1`, sourcing the shell integration simply exits early:
https://github.com/wez/wezterm/blob/69686f4564ea460111936dee3179ecf8c80e9930/assets/shell-integration/wezterm.sh#L11-L25

**Reply by jmpolom** [reply]
This should be documented officially in the docs if it's a feature that users can reliably expect to work and be available for their use.
